### PR TITLE
Remove underspecified and unused Span decorator from Trace SDK.

### PR DESCRIPTION
**Pull request by tsloughter**
## Changes

This removes the "decorate" language from span processors. Technically a breaking change but since consensus seemed to be that something new needed to be designed I figured it best to get it out of the spec first.

Open question if the "for advanced scenarios such as tagging or filtering" part should be left in. I'd say it doesn't apply to custom processors but you technically could use them to do that in OnStart.

### CODE REVIEW

**Review by arminru — COMMENTED** [review]
@tsloughter I remember this was discussed already but I can't find the related issue. Can you please link it to your PR?

**Review by MrAlias — APPROVED** [review]

File: specification/trace/semantic_conventions/instrumentation/aws-lambda.md [hunk 1]
```diff
@@ -66,6 +66,9 @@ contain an incomplete trace context which indicates X-Ray isn’t enabled. The e
 `Context` will be valid and sampled only if AWS X-Ray has been enabled for the Lambda function. A user can
 disable AWS X-Ray for the function if the X-Ray Span Link is not desired.
 
+Implementations MAY provide a configuration option to additionally use the
+`Context` as the parent of the new Span.
+
```
Comment (Aneurysm9) [inline specification/trace/semantic_conventions/instrumentation/aws-lambda.md:71]: Why is this here?  This seems unrelated to the stated purpose of the change.

Comment (tsloughter) [reply specification/trace/semantic_conventions/instrumentation/aws-lambda.md:71]: Bad git stash foo.

**Review by arminru — APPROVED** [review]
LGTM provided @Aneurysm9 comment is addressed. This also looks like a stray change to me.

File: specification/trace/semantic_conventions/instrumentation/aws-lambda.md (hunk 1 above)
Comment (tsloughter) [reply specification/trace/semantic_conventions/instrumentation/aws-lambda.md]: Removed.

**Review by Aneurysm9 — APPROVED** [review]

**Review by reyang — APPROVED** [review]
